### ISSUE TITLE
"statuses" web page uses expensive "keys" command

### ISSUE THREAD

**Issue by benissimo** [utgarda/sidekiq-status#132]
`        namespace_jids = Sidekiq.redis{ |conn| conn.keys('sidekiq:status:*') }`
https://github.com/utgarda/sidekiq-status/blob/v1.0.1/lib/sidekiq-status/web.rb#L78
^ This line triggers the lookup of _all_ keys matching "sidekiq:status:*" in Redis. This is expensive in two ways:

1) The "keys" command is expensive (official docs on Redis.io recommend against using it in production environments, prefer "scan" instead)
2) The results are then paginated anyway, so after asking Redis to sift through all of its keys, find those which match a pattern, sort all of those results, call Sidekiq::Status on each matching key -- after all of this work, all results outside of the current "page" are discarded via `slice`.

Under the current system, when I browsed the "Statuses" page of my Sidekiq web panel, it triggered a long running `keys` command.

**Comment by kenaniah** [comment]
Moving to `SCAN` would require all fields to be sorted within Redis (which would indeed be an improvement). Unfortunately, not all sortable fields are stored within the same struct, which is why the current implementation is in Ruby. 

What you're asking for could be accomplished, but it would require reworking some of the fundamentals of this gem. I'm not opposed to this, but I currently do not have the resources available to make this happen myself. I'd be happy to merge a competent PR though.

**Comment by benissimo** [comment]
How about dropping sort and adding an input to search among the results?

```
SCAN 0 MATCH sidekiq-status:* # default starting point
SCAN 0 MATCH sidekiq-status:*foo* # append search term + another wildcard to filter further
```

Personally I'd gladly give up sorting in order to avoid the risk of taxing Redis with the `KEYS` command.

Rewriting all of this gem's usage of Redis to use sortable sets instead of bare structs could be a great improvement, but that's also a major change (and making it backwards compatible would be a tall order).

Instead swapping SCAN for KEYS and a search box for sorting would be much simpler. Worth pursuing?

**Comment by kenaniah** [comment]
Fixed by #139
